solana-kotlin/src/commonMain/kotlin/net/avianlabs/solana/domain/core/Transaction.kt

```diff
@@ -26,13 +22,9 @@ public open class Transaction internal constructor(
     val serializedMessage = message
       .serialize()
 
-    val signatures = signers.map { signer ->
-      TweetNaCl.Signature.sign(serializedMessage, signer.secretKey).encodeToBase58String()
-    }
-
-    val signatureSet = signatures.toSet()
-    if (signatureSet.size != signatures.size) {
-      logger.warn { "Duplicate signatures detected" }
+    val signatures = signers.associate { signer ->
+      signer.publicKey to
+        TweetNaCl.Signature.sign(serializedMessage, signer.secretKey)
```

Comment:
Why do we do this now ?

Reply:
so the signatures can preserve the order of account meta list